processor/src/host/advice/mod.rs

```diff
@@ -152,6 +152,11 @@ impl AdviceProvider {
     // ADVICE MAP
     // --------------------------------------------------------------------------------------------
 
+    /// Returns a reference to the inner [`AdviceMap`].
+    pub fn advice_map(&self) -> &AdviceMap {
```

Comment:
Do we still ned `contains_map_key()` and `get_mapped_values()` then if we expose `&AdviceMap`?

Reply:
Yeah, if we have the `advice_map()` accessor, I don't really see a need for them.